### PR TITLE
Improve WG Key Management

### LINKED ISSUE #15: Improve WG Key Management
## Description
WG provides key pair generation capability. Integrate this into the system better.

## Acceptance Criteria
- [x] `GUILD_PRIVATE_KEY` is optional.
- [x] If `GUILD_PRIVATE_KEY` not provided, use the one in the current config. If not provided and there is no config, generate new key pair.
- [x] If `GUILD_PRIVATE_KEY` is provided, generate matching public key, even in the case of an existing config.
- [x] Remove `GUILD_PUBLIC_KEY` from .env and generate using `wg pubkey` and the `GUILD_PRIVATE_KEY` .

### PR DESCRIPTION
**Pull request by ajmassi**
### AC
* Add more responsive key-pair behavior based on `.env`-provided private key, and presence of existing WireGuard config file
* Removed `GUILD_PUBLIC_KEY` from environment variables parsed by Settings - now generated dynamically with or from `Settings.guild_private_key`

### Other
* Rename `config.py` to `settings.py`, the BaseSettings instance generated by this file has similarly been renamed from `conf` to `settings`
* Add tests to reflect new key pair behaviors
* Fix bug where reloading an existing wg.conf would wipe contents
* Add `make test` shortcut for launching pytest
* Add coverage for pytest

### CODE REVIEW

File: tests/test.env [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Discord
+BOT_TOKEN=93a8377394f953ce692f3ab61aea8cd7cb37102ebc2fca058329c6d26cad85190c08f9b28149b79ed2d989c0f13516a4c48e806a1ac657a20cc0f39010f8d8c3
```
Comment (abefleming) [inline tests/test.env:2]: Is this intended?

Comment (abefleming) [reply tests/test.env:2]: I presume its a test token, but looks pretty convincing to me.

**Review by abefleming — CHANGES_REQUESTED** [review]

File: tests/test.env (hunk 1 above)
Comment (ajmassi) [reply tests/test.env:2]: Lol yea it's garbage, just a default to pass validators
Couldnt come up with something cute/obvious so I generated junk

Comment (abefleming) [reply tests/test.env:2]: My bad, I didn't realize this was a test file. 😅

**Review by abefleming — APPROVED** [review]

**Pull request merged**